### PR TITLE
Adds no-unhandled-errors rule and fixes yield-effects issue

### PR DESCRIPTION
**Pull request by burabure**

### CODE REVIEW

File: lib/rules/no-unhandled-errors.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    },
+    schema: []
+  },
+
+  create: function(context) {
+    var effectLocalNames = []
+    var effectImportedNames = []
+    return {
+      "ImportDeclaration": function(node) {
+        if (/^redux-saga(\/effects)?/.test(node.source.value)) {
```
Comment (pke) [inline lib/rules/no-unhandled-errors.js]: Nice catch. Maybe we should factor that out to a util function?

Comment (burabure) [inline lib/rules/no-unhandled-errors.js]: Agree, done =)

**Comment by pke** [comment]
@burabure Thanks for contributing!

**Comment by pke** [comment]
Upon further thinking I came to the conclusion that this rule does not make much sense, or at least should not be `recommended`.

Consider the following saga:

```es6
function* refreshInvoices(access_token) {
  while (true) {
    try {
      let invoices = yield call(api.getInvoicesAsync, access_token)
      yield put(actions.invoicesRefreshed(invoices))
      yield take(actions.REFRESH_INVOICES)
    } catch (error) {
      console.error(error)
    }
  }
}
```

The saga like this would request invoices over and over from the API when an error occurs in the sagas flow.

However, this flow would stop the saga after one run and wait for a `REFRESH_INVOICES` to give the server some slack.

```es6
function* refreshInvoices(access_token) {
  while (true) {
    try {
      let invoices = yield call(api.getInvoicesAsync, access_token)
      __DEV__ && console.info(`${invoices.length} invoices fetched`)
      yield put(actions.invoicesRefreshed(invoices))
    } catch (error) {
      console.error(error)
    }
    yield take(actions.REFRESH_INVOICES)
  }
}
```

What is your take on this? Of course, one could wrap all this in multiple try/catch but that needlessly complicates the whole saga.

**Comment by burabure** [comment]
@pke you need a `finally` there ;D
```js
function* refreshInvoices(access_token) {
  while (true) {
    try {
      let invoices = yield call(api.getInvoicesAsync, access_token)
      __DEV__ && console.info(`${invoices.length} invoices fetched`)
      yield put(actions.invoicesRefreshed(invoices))
    } catch (error) {
      console.error(error)
    } finally {
      yield take(actions.REFRESH_INVOICES)
    }
  }
}
```

**Comment by pke** [comment]
Yeah just when discussing that with my colleague I had the same idea :)

**Pull request merged**